purchase_order_generator/models/purchase_order_generator.py

```diff
@@ -84,8 +84,17 @@ def generate_cpo(self):
                 [
                     ("product_tmpl_id", "=", product_id),
                     ("name", "=", supplier.id),
-                ]
+                ],
+                order="date_start desc",
+                limit=1,
             )
+
+            if not supplierinfo:
+                product_name = self.env["product.template"].browse(product_id)
+                raise ValidationError(
+                    _("No supplier defined for product %s") % product_name
```

Comment:
you should give the translated product name because they manly duplicate it and so only the translation is correct while the product name is the one from the duplicated name (with copy mentionned).

Reply:
It's ok, we talked about it.